lib/connection/connection.js

```diff
@@ -103,54 +103,39 @@ function Connection(context)
     return id;
   };
 
-  this.heartbeat = function (self)
+  this.heartbeat = callback =>
   {
     Logger.getInstance().debug("Issuing heartbeat call");
-    var requestID = uuidv4();
+    const requestID = uuidv4();
 
-      const request = services.sf.request(
+    services.sf.request(
         {
           method: 'POST',
           url: Url.format(
-            {
-              pathname: '/session/heartbeat',
-              search: QueryString.stringify(
-                {
-                  requestId: requestID
-                })
-            }),
-          callback: function (err, body)
+              {
+                pathname: '/session/heartbeat',
+                search: QueryString.stringify(
+                    {
+                      requestId: requestID
+                    })
+              }),
+          callback: Util.isFunction(callback) ? callback : function (err, body)
           {
             if (err) {
-              console.error("Error issuing heartbeat call : " + err.message);
+              Logger.getInstance().error('Error issuing heartbeat call: %s', err.message);
               throw err;
             }
-            Logger.getInstance().debug("heartbeat response %s", body);
+            Logger.getInstance().debug("Heartbeat response %s", JSON.stringify(body));
```

Comment:
Why keep double quotes here?

Reply:
backtick is used for string templating, but single and double quotes are the same - just pick one convention

I changed double quote to single in touched code